src/Repository/Doctrine/UserRepository.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Copyright (C) 2018 Adshares sp. z. o.o.
+ *
+ * This file is part of ADS Operator
+ *
+ * ADS Operator is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * ADS Operator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ADS Operator.  If not, see <https://www.gnu.org/licenses/>
+ */
+
+namespace Adshares\AdsOperator\Repository\Doctrine;
+
+use Adshares\AdsOperator\Document\User;
+use Adshares\AdsOperator\Repository\UserRepositoryInterface;
+use Doctrine\ODM\MongoDB\DocumentRepository;
+
+class UserRepository extends DocumentRepository implements UserRepositoryInterface
+{
+
+    public function findByEmail(string $email): ?User
+    {
+        /** @var User $user */
+        $user = $this->findOneBy(['email' => $email]);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$user". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1488)

Reply:
:+1:

Reply:
It has to be here regarding to `phpstan` checker and the fact that `findOneBy` returns `object`.